shell/platform/windows/keyboard_manager_win32.cc

```diff
@@ -0,0 +1,208 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <assert.h>
+#include <string>
+
+#include "keyboard_manager_win32.h"
+
+#include "keyboard_win32_common.h"
+
+namespace flutter {
+
+static char32_t CodePointFromSurrogatePair(wchar_t high, wchar_t low) {
+  return 0x10000 + ((static_cast<char32_t>(high) & 0x000003FF) << 10) +
+         (low & 0x3FF);
+}
+
+static uint16_t ResolveKeyCode(uint16_t original,
+                               bool extended,
+                               uint8_t scancode) {
+  switch (original) {
+    case VK_SHIFT:
+    case VK_LSHIFT:
+      return MapVirtualKey(scancode, MAPVK_VSC_TO_VK_EX);
+    case VK_MENU:
+    case VK_LMENU:
+      return extended ? VK_RMENU : VK_LMENU;
+    case VK_CONTROL:
+    case VK_LCONTROL:
+      return extended ? VK_RCONTROL : VK_LCONTROL;
+    default:
+      return original;
+  }
+}
+
+static bool IsPrintable(uint32_t c) {
+  constexpr char32_t kMinPrintable = ' ';
+  constexpr char32_t kDelete = 0x7F;
+  return c >= kMinPrintable && c != kDelete;
+}
+
+KeyboardManagerWin32::KeyboardManagerWin32(WindowDelegate* delegate)
+    : window_delegate_(delegate) {}
+
+bool KeyboardManagerWin32::HandleMessage(UINT const message,
+                                         WPARAM const wparam,
+                                         LPARAM const lparam) {
+  switch (message) {
```

Comment:
This switch case is almost unchanged from what was in `window_win32.cc`, except that `return 0` statements are changed to `return true` due to different definition of the return value.